src/transformers/utils/tensor_parallel_utils.py

```diff
@@ -0,0 +1,196 @@
+# coding=utf-8
+# Copyright 2021 TUNiB Inc and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from transformers import (
+    BertPreTrainedModel,
+    GPT2PreTrainedModel,
+    TransfoXLPreTrainedModel,
+    T5PreTrainedModel,
+    RobertaPreTrainedModel,
+)
+
+
+"""
+All the mapping for tensor parallelism.
+This mapping is following the follow format.
+
+TENSOR_PARALLEL_MAPPING = {
+    PreTrainedModel class: {
+        "col": list of column parallel parameters,
+        "row": list of row parallel parameters,
+        "update": list of attributes to be updated,
+        "col_no_replacement": list of column parallel parameters without module replacement (opt)
+        "row_no_replacement": list of row parallel parameters without module replacement (opt),
+        ...
+        could be added more to avoid exceptions.
+    }
+}
+
+Or if a model A has the same map with the other model B, define like:
+
+TENSOR_PARALLEL_MAPPING = {
+    PreTrainedModel class A: PreTrainedModel class B
+}
+
+Then, call ``copy_mapping(PreTrainedModel class A)``.
+"""
+
+TENSOR_PARALLEL_MAPPING = {
+    BertPreTrainedModel: {
+        "col": ["query", "key", "value", "intermediate.dense"],
+        "row": ["output.dense"],
+        "update": ["num_attention_heads", "all_head_size"],
+    },
+    GPT2PreTrainedModel: {
+        "col": ["c_attn", "q_attn", "c_fc"],
+        "row": ["c_proj"],
+        "update": ["embed_dim", "split_size", "num_heads"],
+    },
+    T5PreTrainedModel: {
+        "col": ["Attention.q", "Attention.k", "Attention.v", "DenseReluDense.wi"],
+        "row": ["Attention.o", "DenseReluDense.wo"],
+        "row_no_replacement": ["relative_attention_bias"],
+        "update": ["d_model", "n_heads", "inner_dim"],
+    },
+    RobertaPreTrainedModel: BertPreTrainedModel,
```

Comment:
I added the `copy_mapping` because there are some models that have same mapping in the Transformers.
In this case, we don't need to copy the mapping.